src/components/CalendarYearGrid.jsx

```diff
@@ -0,0 +1,161 @@
+import React, { PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+import shallowCompare from 'react-addons-shallow-compare';
+import momentPropTypes from 'react-moment-proptypes';
+import moment from 'moment';
+import cx from 'classnames';
+
+import CalendarYear from './CalendarYear';
+
+import isTransitionEndSupported from '../utils/isTransitionEndSupported';
+import getTransformStyles from '../utils/getTransformStyles';
+
+import OrientationShape from '../shapes/OrientationShape';
+
+import { HORIZONTAL_ORIENTATION, VERTICAL_ORIENTATION } from '../../constants';
+
+const propTypes = {
+  initialYear: momentPropTypes.momentObj,
+  firstVisibleYearIndex: PropTypes.number,
+  isAnimating: PropTypes.bool,
+  numberOfYears: PropTypes.number,
+  modifiers: PropTypes.object,
```

Comment:
As I understandthe code, the modifiers use a dynamic key to apply that key as a class name to the css, so I think it can't be a shape. Is that right?